graspy/subgraph/sg.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+from scipy.stats import fisher_exact
+
+
+class SignalSubgraph:
+    """
+    Estimate the signal-subgraph of a set of labeled graph samples.
+    
+    The incoherent estimator finds the signal-subgraph, constrained by the number of edges.
+    The coherent estimator finds the signal-subgraph, constrained by the number of edges and by the number of vertices that the edges in the signal-subgraph may be incident to.
+
+    Parameters
+    ----------
+    graphs: array-like, shape (n_vertices, n_vertices, s_samples)
+        A series of labeled (n_vertices, n_vertices) unweighted graph samples. If undirected, the upper or lower triangle matrices should be used.
+        
+    labels: vector, length (s_samples)
+        A vector of class labels. There must be a maximum of two classes.
+         
+    Attributes
+    ----------
+    contmat: array-like, shape (n_vertices, n_vertices, 2, 2)
+        An array that stores the 2-by-2 contingency matrix for each point in the graph samples.
+    sigsub: tuple, shape (2, n_edges)
+        A tuple of a row index array and column index array, where n_edges is the size of the signal-subgraph determined by *constraints*.
+    
+    References
+    ----------
+    .. [1] J. T. Vogelstein, W. R. Gray, R. J. Vogelstein, and C. E. Priebe, "Graph Classification using Signal-Subgraphs: Applications in Statistical Connectomics," arXiv:1108.1427v2 [stat.AP], 2012.
+    
+    """
+
+    def __init__(self, graphs, labels):
```

Comment:
I would have no inputs for class constructor here. graphs and labels should go to fit and fit_transform method

Reply:
You can either abstract out the type checking to a separate private function, or just move it to fit

Reply:
So if I'm moving type checking to fit, and inputs to fit and fit_transform, is it convention to have an empty constructor? Or should I get rid of __init__ altogether